src/variable.cpp

```diff
@@ -77,6 +87,36 @@ void Variable::swap(Variable &rhs)
     std::swap(this->mPimpl, rhs.mPimpl);
 }
 
+void Variable::addEquivalence(const VariablePtr &equivalentVariable)
+{
+    setEquivalentTo(equivalentVariable);
+    VariablePtr *pV = new VariablePtr(this);
```

Comment:
I'm starting to wonder if, given the reciprocal nature of this method, perhaps it is better implemented as a static method on the variable class that takes two `VariablePtr` arguments?